cfme/tests/cloud_infra_common/test_vm_instance_analysis.py

```diff
@@ -0,0 +1,590 @@
+
+    ssa_vm.smartstate_scan(check_task_result=True)
+
+    # Check that all data has been fetched
+    current = ssa_vm.get_detail(properties=('Configuration', 'Files'))
+    assert current != '0', "No files were scanned"
+
+    details_property_view = ssa_vm.open_details(("Configuration", "Files"))
+    try:
+        details_property_view.paginator.find_row_on_pages(
+            details_property_view.table, name=ssa_expect_files[0])
+    except NoSuchElementException:
+        pytest.fail('File {} was not found in details table after SSA run'.format(
+            ssa_expect_files[0]))
+
+
+@pytest.mark.tier(2)
+@pytest.mark.long_running
+def test_drift_analysis(request, ssa_vm, soft_assert, appliance, ssa_profile):
+    """ Tests drift analysis is correct
+
+    Metadata:
+        test_flag: vm_analysis
+    """
+
+    ssa_vm.load_details()
+    drift_num_orig = 0
+    drift_orig = ssa_vm.get_detail(properties=("Relationships", "Drift History"))
+    if drift_orig != 'None':
+        drift_num_orig = int(drift_orig)
+    ssa_vm.smartstate_scan(check_task_result=True)
+    wait_for(
+        lambda: ssa_vm.get_detail(
+            properties=("Relationships", "Drift History")) == str(drift_num_orig + 1),
+        delay=20,
+        num_sec=360,
+        message="Waiting for Drift History count to increase",
+        fail_func=appliance.server.browser.refresh
+    )
+    drift_new = int(ssa_vm.get_detail(properties=("Relationships", "Drift History")))
+
+    # add a tag and a finalizer to remove it
+    ssa_vm.add_tag('Department', 'Accounting')
+    request.addfinalizer(lambda: ssa_vm.remove_tag('Department', 'Accounting'))
+    ssa_vm.smartstate_scan(check_task_result=True)
+    wait_for(
+        lambda: ssa_vm.get_detail(
+            properties=("Relationships", "Drift History")) == str(drift_new + 1),
+        delay=20,
+        num_sec=360,
+        message="Waiting for Drift History count to increase",
+        fail_func=appliance.server.browser.refresh
+    )
+    # check drift difference
+    soft_assert(ssa_vm.equal_drift_results('Department (1)', 'My Company Tags', 0, 1),
+                "Drift analysis results are equal when they shouldn't be")
+
+    # Test UI features that modify the drift grid
+    drift_analysis_view = appliance.browser.create_view(DriftAnalysis)
+
+    # Accounting tag should not be displayed, because it was changed to True
+    drift_analysis_view.toolbar.same_values_attributes.click()
+    soft_assert(
+        not drift_analysis_view.drift_analysis.check_section_attribute_availability(
+            'Department (1)'), "Department (1) row should be hidden, but not")
+
+    # Accounting tag should be displayed now
+    drift_analysis_view.toolbar.different_values_attributes.click()
+    soft_assert(
+        drift_analysis_view.drift_analysis.check_section_attribute_availability('Department (1)'),
+        "Department (1) row should be visible, but not")
```

Comment:
You moved the entire file into another directory. It's not clear now which changes were made by you.

Reply:
Yes I understand, but I have to move it as these tests covers infra and cloud parts. As for changes, the tests where updated regards all widgetastic changes, and the number of cases didn't change